src/Stryker.Core/Stryker.Core/Reporters/JsonReporter.cs

```diff
@@ -106,22 +284,29 @@ public static JsonReportComponent FromProjectComponent(IReadOnlyInputComponent c
 
                 if (component is FolderComposite folderComponent)
                 {
+                    report.Name = component.Name is null ? options.ProjectUnderTestNameFilter : folderComponent.RelativePath;
+                    report.ChildResults = new List<JsonReportComponent>();
+
                     foreach (var child in folderComponent.Children)
                     {
-                        report.ChildResults.Add(FromProjectComponent(child, thresholdOptions));
+                        report.ChildResults.Add(FromProjectComponent(child, options));
                     }
                 }
                 else if (component is FileLeaf fileComponent)
                 {
+                    report.Name = fileComponent.Name;
                     report.Source = fileComponent.SourceCode;
+                    report.Language = fileComponent.Mutants.First().Mutation.OriginalNode.Language;
```

Comment:
This could possibly break when no mutations are made during a stryker run. Also I'm not comfortable with the trust in the Language propery on the node. We cannot be sure this will always be "sharp". For me a string literal would be fine here.